SparseGrids/tasgridWrapper.cpp

```diff
@@ -362,6 +379,13 @@ void TasgridWrapper::createQuadrature(){
         cerr << "ERROR: createQuadrature" << endl;
     }
 }
+void TasgridWrapper::createExoticQuadrature(){
+    TasGrid::TasmanianSparseGrid weight_surrogate;
+    weight_surrogate.read(weightfilename.c_str());
+    // Depth = 2 * level - 1.
+    int level = depth % 2 == 1 ? (depth + 1) / 2 : depth / 2 + 1;
```

Comment:
Why is level different from depth?

Reply:
From what I understand, depth refers to (roughly) the number of points that make up the grid. For the exotic quadrature function though, level is the number of Gauss-Legendre correction nodes or the generated weighted Gauss nodes.

For example, if we wanted an exotic quadrature with ~15 nodes, we would want to generate 8 Gauss-Legendre nodes and 8 weighted Gauss nodes (using `decompose()`) and account for the fact that `0.0` might be common between these nodes. Hence, the "level" given to `getExoticQuadrature()` would be 8 instead of the desired depth of 15.

Reply:
Sparse Grid logic seldom goes by "number of points", we usually look at "level" or "polynomial order".
For the exotic quadrature, we should go by "level = depth".

An alternative could be that we add the criteria of polynomial order, i.e., find the level that matches the order but we can leave that off for now.

On a side note, even if you want to go by the number of points, there's no way to guarantee that some of the points for the shifted weight would not match the Gauss-Legendre points, so the formula may be wrong. If you want to specify the points exactly, you will have to change the logic of the construction and stop the find-roots process when you have enough total points. Let's not do that.

Reply:
Gotcha. I changed level to be the same as depth in the latest push.